exonum-java-binding-core/rust/integration_tests/tests/transaction_proxy.rs

```diff
@@ -84,20 +91,51 @@ fn execute_should_panic_if_java_error_occurred() {
 }
 
 #[test]
-fn execute_should_return_err_if_java_exception_occurred() {
-    let invalid_tx =
+#[should_panic(expected = "Java exception: java.lang.ArithmeticException")]
+fn execute_should_panic_if_java_exception_occurred() {
+    let panic_tx =
         create_throwing_mock_transaction_proxy(EXECUTOR.clone(), ARITHMETIC_EXCEPTION_CLASS);
     let db = MemoryDB::new();
     let mut fork = db.fork();
+    panic_tx.execute(&mut fork).unwrap();
+}
+
+#[test]
+fn execute_should_return_err_if_tx_exec_exception_occurred() {
+    let err_code: i8 = 1;
+    let err_message = "Expected exception";
+    let invalid_tx = create_throwing_exec_exception_mock_transaction_proxy(
+        EXECUTOR.clone(),
+        false,
+        err_code,
+        err_message,
+    );
+    let db = MemoryDB::new();
+    let mut fork = db.fork();
     let err = invalid_tx
         .execute(&mut fork)
         .map_err(TransactionError::from)
         .expect_err("This transaction should be executed with an error!");
-    assert!(
-        err.description()
-            .unwrap()
-            .starts_with("Java exception: java.lang.ArithmeticException",)
+    assert!(err.description().unwrap().starts_with(err_message));
```

Comment:
We must assert that the err includes the code — that's the feature we are testing.

Reply:
one of the features :-)